### ISSUE TITLE
Main-process crash (EXC_BAD_ACCESS / heap UAF) when destroying a <webview> with an attached webContents.debugger session

### ISSUE THREAD

**Issue by regrlomon** [electron/electron#53819]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

43.1.1

### What operating system(s) are you using?

macOS

### Operating System Version

macOS arm64 (OS X 26.4.1, 25E253)

### What arch are you using?

x64

### Does the issue also appear in Chromium / Google Chrome?

I don't know how to test

### Expected Behavior

Destroying (unmounting from the DOM) a <webview> tag whose guest webContents has a webContents.debugger CDP session attached should never crash the main (browser) process. Detaching the debugger before destroying the webview, with arbitrary delay, should make it safe.

### Actual Behavior

The **main process crashes** (whole app dies) when a `<webview>` is destroyed while/after a CDP session was attached to it. Three distinct crash signatures observed, all on `Thread 0 CrBrowserMain`:

**1) Null-deref in `RenderFrameDevToolsAgentHost` during navigation notifications** (observed both in production and in the minimal repro):
```
0  Electron Framework!content::RenderFrameDevToolsAgentHost::UpdateFrameHost(content::RenderFrameHostImpl*) [render_frame_devtools_agent_host.cc:1082]   (EXC_BAD_ACCESS, addr 0x28)
1  Electron Framework!content::RenderFrameDevToolsAgentHost::ReadyToCommitNavigation(content::NavigationHandle*) [:518]
2  Electron Framework!content::WebContentsImpl::WebContentsObserverList::NotifyObservers<...>
3  Electron Framework!content::WebContentsImpl::ReadyToCommitNavigation(content::NavigationHandle*) [web_contents_impl.cc:7761]
4  Electron Framework!content::NavigationRequest::ReadyToCommitNavigation(bool)
5  Electron Framework!content::NavigationRequest::CommitNavigation() [navigation_request.cc:6915]
```
Variant: same class, `OnNavigationRequestWillBeSent` (render_frame_devtools_agent_host.cc:748, addr 0x20).

**2) Heap use-after-free in `DevToolsSession` message dispatch** (register `x1 = 0xcdcdcdcdcdcdcdcd` — destroyed-object fill pattern):
```
0  Electron Framework!content::DevToolsSession::DispatchProtocolResponseOrNotification(...) [devtools_session.cc:602]   (EXC_BAD_ACCESS)
1  Electron Framework!content::DevToolsSession::DispatchProtocolNotification(...)
```

**3) `CHECK` failure while re-launching the guest renderer process after remount** (SIGTRAP / EXC_BREAKPOINT):
```
0  Electron Framework!electron::ElectronBrowserClient::AppendExtraCommandLineSwitches(...) [immediate_crash.h:180]
1  Electron Framework!content::RenderProcessHostImpl::AppendRendererCommandLine [render_process_host_impl.cc:3714]
2  Electron Framework!content::RenderProcessHostImpl::Init() [render_process_host_impl.cc:2041]
3  Electron Framework!content::RenderFrameHostManager::InitRenderView(...)
4  Electron Framework!content::WebContentsImpl::AttachInnerWebContentsImpl(...) [web_contents_impl.cc:3365]
5  Electron Framework!electron::WebViewGuestDelegate::AttachToIframe(...)
```

## How to reproduce

Two files, no build step, no external dependencies beyond `electron@43.1.1` itself.

**`main.js`**
```js
const { app, BrowserWindow, crashReporter } = require('electron');
crashReporter.start({ submitURL: 'http://127.0.0.1', uploadToServer: false });

const HAMMER_MS = 20;    // CDP command interval, mimics an automation agent
const REMOUNT_MS = 1500; // destroy & recreate the <webview>, mimics a React remount

app.whenReady().then(() => {
  const win = new BrowserWindow({
    width: 960, height: 720,
    webPreferences: { webviewTag: true },
  });
  win.loadFile('host.html');

  // attach a debugger to every guest webContents and keep a high CDP message rate
  app.on('web-contents-created', (_e, contents) => {
    if (contents.getType() !== 'webview') return;
    try { contents.debugger.attach('1.3'); } catch {}
    let n = 0;
    const timer = setInterval(() => {
      if (contents.isDestroyed()) { clearInterval(timer); return; }
      if (!contents.debugger.isAttached()) { try { contents.debugger.attach('1.3'); } catch { return; } }
      const p = (n % 10 === 0)
        ? contents.debugger.sendCommand('Runtime.enable')
        : contents.debugger.sendCommand('Runtime.evaluate', { expression: '1+1' });
      p.catch(() => {});
      n++;
    }, HAMMER_MS);
    contents.on('destroyed', () => clearInterval(timer));
  });
});

setTimeout(() => app.exit(0), 90000).unref();
app.on('window-all-closed', () => app.quit());
```

**`host.html`**
```html
<!doctype html>
<html><body style="font:13px/1.5 monospace;margin:8px">
  <div id="stat">remount #0</div>
  <webview id="wv" src="https://www.zhipin.com/" style="width:900px;height:560px"></webview>
  <script>
    // destroy & recreate the webview every 1.5s (like a React remount)
    let n = 0;
    setInterval(function () {
      const old = document.getElementById('wv');
      const fresh = document.createElement('webview');
      fresh.id = 'wv';
      fresh.src = 'https://www.zhipin.com/';
      fresh.style.cssText = 'width:900px;height:560px';
      old.replaceWith(fresh);
      n++;
      document.getElementById('stat').textContent = 'remount #' + n;
    }, 1500);
  </script>
</body></html>
```

Run:
```
npm init -y && npm i electron@43.1.1
npx electron main.js
```

Crash usually occurs within 10–20 remount cycles (15–30 s). Reproduces faster with a content-heavy site on a slower network (wide navigation-timing window); a lightweight page (e.g. example.com) may survive many more cycles, but the mechanism is identical.

## Control experiments (all on the same page/network)

| Setup | Result |
|---|---|
| `BrowserWindow` + attached debugger + `reload()` every 3 s (webContents never destroyed) | **No crash** (1649 CDP commands, 13 reloads) |
| `<webview>` + attached debugger + repeated navigation, webview never destroyed | **No crash** (22 reloads) |
| Reproducer as above (webview destroyed & recreated) | **Crash** (all 3 signatures above) |
| Reproducer + `debugger.detach()` on all guests right before destroying | **Still crashes** |
| Reproducer + `detach()` + 500 ms delay before destroying | **Still crashes** (7 remounts) |

This isolates the crash to the **combination of "CDP session was attached" + "webview element destroyed"**: navigation traffic and CDP load alone are safe; destroying a webview that ever had a session attached is not — and JS-level detach (even with delay) does not make destruction safe, which suggests the internal DevTools session teardown (mojo) is not synchronized with the guest teardown in the browser process.

**Comment by ayusharora07** [comment]
Hi @regrlomon, thanks for the detailed report and minimal reproducer!

I'm taking a look into this issue to debug the teardown lifecycle desynchronization between WebViewGuestDelegate and DevToolsSession / RenderFrameDevToolsAgentHost. I'll update here once I have a fix or further findings
